R/derive_merged.R

```diff
@@ -0,0 +1,708 @@
+#' Merge Variables from a Dataset to the Input Dataset
+#'
+#' Merge variables from a dataset to the input dataset. The observations to
+#' merge can be selected by a condition and/or selecting the first or last
+#' observation for each by group.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `by_vars` parameter are expected.
+#'
+#' @param dataset_add Additional dataset
+#'
+#'   The variables specified by the `by_vars`, the `new_vars`, and the `order`
+#'   parameter are expected.
+#'
+#' @param order Sort order
+#'
+#'   The first or last observation from the addtional dataset is selected with
+#'   respect to the specified order.
+#'
+#'   *Permitted Values*: list of variables or `desc(<variable>)` function calls
+#'
+#' @param new_vars Variables to add
+#'
+#'   The specified variables are added to the output dataset. Variables can be
+#'   renamed by naming the element, i.e., `new_vars = vars(<new name> = <old
+#'   name>)`.
+#'
+#'   If the parameter is not specified or set to `NULL`, all variables from the
+#'   additional dataset are added.
+#'
+#'   *Permitted Values*: list of variables created by `vars()`
+#'
+#' @param mode Selection mode
+#'
+#'   Determines if the first or last observation is selected.
+#'
+#'   *Permitted Values*: `"first"`, `"last"`
+#'
+#' @param by_vars Grouping variables
+#'
+#'   The input dataset and the selected observations from the additional dataset
+#'   are merged by the specified by variables. The by variables must be a unique
+#'   key of the selected observations.
+#'
+#'   *Permitted Values*: list of variables
+#'
+#' @param filter_add Filter for additional data
+#'
+#'   Only observations fulfilling the specified condition are taken into account
+#'   for merging. If the parameter is not specified, all observations are
+#'   considered.
+#'
+#'   *Permitted Values*: a condition
+#'
+#' @param match_flag Match flag
+#'
+#'   If the parameter is specified, the specified variable is added to the
+#'   output dataset and set to `TRUE` for all by groups with an observation in
+#'   the additional dataset. For all other by groups it is set to `NA`.
+#'
+#'   *Permitted Values*: Variable name
+#'
+#' @param check_type Check uniqueness?
+#'
+#'   If `"warning"` or `"error"` is specified, the specified message is issued
+#'   if the observations of the (restricted) additional dataset are not unique
+#'   with respect to the by variables and the order.
+#'
+#'   *Permitted Values*: `"none"`, `"warning"`, `"error"`
+#'
+#' @param duplicate_msg Message of unique check
+#'
+#'   If the uniqueness check fails, the specified message is displayed.
+#'
+#'   *Default*:
+#'
+#'   ```
+#'   paste("Dataset `dataset_add` contains duplicate records with respect to",
+#'         enumerate(vars2chr(by_vars))
+#'   ```
+#'
+#' @details
```

Comment:
Can we add some clarification this is essentially a left join (e.g. all records from the left `dataset` are kept and records from the right `dataset_add` are not included if there are no matching records in the left `dataset`)?

Reply:
I have added more details. Is it clear now?

Reply:
Yes, thanks!